x-pack/plugin/text-structure/qa/text-structure-with-security/build.gradle

```diff
@@ -0,0 +1,31 @@
+apply plugin: 'elasticsearch.yaml-rest-test'
+
+dependencies {
+  yamlRestTestImplementation project(path: xpackModule('core'))
+  yamlRestTestImplementation project(path: xpackModule('core'), configuration: 'testArtifacts')
+  yamlRestTestImplementation project(path: xpackProject('plugin').path, configuration: 'testArtifacts')
+}
+
+// bring in text structure rest test suite
+restResources {
+  restApi {
+    includeCore '_common', 'cluster', 'nodes', 'indices', 'index', 'search', 'get', 'count', 'ingest', 'bulk'
+    includeXpack 'text_structure'
+  }
+  restTests {
+    includeXpack 'text_structure'
+  }
+}
+
+tasks.named("yamlRestTest").configure {
+}
+
+testClusters.all {
+  testDistribution = 'DEFAULT'
+  extraConfigFile 'roles.yml', file('roles.yml')
+  user username: "x_pack_rest_user", password: "x-pack-test-password"
+  user username: "text_structure_admin", password: "x-pack-test-password", role: "minimal,monitor_text_structure"
```

Comment:
I would call this user `text_structure_user`.  The general mapping is that `manage*` privileges are considered "admin" and `monitor*` privileges are considered "user".

Reply:
++